core/src/main/java/fj/data/List.java

```diff
@@ -1840,4 +1839,31 @@ private void copy() {
     @Override public String toString() {
         return Show.listShow( Show.<A>anyShow() ).show( this ).foldLeft((s, c) -> s + c, "" );
     }
+
+    /**
+     * True if the list is a singleton.  Faster than checking <code>length() == 1</code> for any
+     * list with <code>length() > 1</code>.
+     */
+	public boolean isSingle() {
+		return isNotEmpty() && tail().isEmpty();
+	}
+
+	/**
+	 * Compare elements of this list to the elements in the other list.  If one
+	 * list is a prefix of the other, the shorter list is considered to be less
+	 * than the longer list.
+	 *
+	 * @param ord Ordering to use for comparison
+	 * @param other Other list to compare with
+	 * @return An Ordering result
+	 */
+	public Ordering compare(Ord<A> ord, List<A> other) {
+		Ordering cmp = Ord.booleanOrd.compare(isEmpty(), other.isEmpty());
+		if(isEmpty() || cmp != Ordering.EQ)
+			return cmp;
+		cmp = ord.compare(head(), other.head());
+		if(cmp != Ordering.EQ)
+			return cmp;
+		return tail().compare(ord, other.tail());
+	}
```

Comment:
You may not be aware, but there are static fields on Ord for many of the common classes
Reusing this implementation means the method body is:
```
return Ord.listOrd(ord).compare(this, other);
```

Probably best just to reuse (most of) the javadoc from Ord.listOrd.